qiskit_experiments/library/stark/ramsey_xy.py

```diff
@@ -0,0 +1,319 @@
+        options.update_options(
+            stark_amp=0.0,
+            stark_channel=None,
+            stark_freq_offset=80e6,
+            stark_sigma=15e-9,
+            stark_risefall=2,
+            min_freq=5e6,
+            max_freq=100e6,
+            delays=None,
+        )
+        options.set_validator("stark_freq_offset", (0, np.inf))
+        options.set_validator("stark_channel", pulse.channels.PulseChannel)
+        return options
+
+    def _set_backend(self, backend: Backend):
+        super()._set_backend(backend)
+        self._timing = BackendTiming(backend)
+
+    def delays(self) -> np.ndarray:
+        """Delay values to use in circuits.
+
+        .. note::
+
+            The delays are computed with min_freq and max_freq experiment options.
+            The maximum point is computed from the min_freq to guarantee the result
+            contains at least one Ramsey oscillation cycle at this frequency.
+            The interval is computed from the max_freq to sample at frequency
+            higher than the Nyquist frequency of this frequency.
+
+        Returns:
+            The list of delays to use for the different circuits based on the
+            experiment options.
+        """
+        opt = self.experiment_options  # alias
+
+        if opt.delays is None:
+            # Delay is longer enough to capture 1 cycle of the minmum frequency.
+            # Fitter can still accurately fit samples shorter than 1 cycle.
+            max_period = 1 / opt.min_freq
+            # Inverse of interval should be greater than Nyquist frequency.
+            sampling_freq = 2 * opt.max_freq
+            interval = 1 / sampling_freq
+            return np.arange(0, max_period, interval)
+        return opt.delays
+
+    def parameterized_circuits(self) -> Tuple[circuit.QuantumCircuit, circuit.QuantumCircuit]:
+        """Create circuits with parameters for Ramsey XY experiment with Stark tone.
+
+        Returns:
+            Quantum template circuits for Ramsey X and Ramsey Y experiment.
+        """
+        opt = self.experiment_options  # alias
+        param = circuit.Parameter("delay")
+
+        # Pulse gates
+        stark_v = circuit.Gate("StarkV", 1, [])
+        stark_u = circuit.Gate("StarkU", 1, [param])
+
+        # Note that Stark tone yields negative (positive) frequency shift
+        # when the Stark tone frequency is higher (lower) than qubit f01 frequency.
+        # This choice gives positive frequency shift with positive Stark amplitude.
+        qubit_f01 = self._backend_data.drive_freqs[self.physical_qubits[0]]
+        stark_freq = qubit_f01 - np.sign(opt.stark_amp) * opt.stark_freq_offset
+        stark_amp = np.abs(opt.stark_amp)
+        stark_channel = opt.stark_channel or pulse.DriveChannel(self.physical_qubits[0])
+        ramps_dt = self._timing.round_pulse(time=2 * opt.stark_risefall * opt.stark_sigma)
+        sigma_dt = ramps_dt / 2 / opt.stark_risefall
```

Comment:
Rounding risefall is only option. Note that StarkV and U must have the identical edge shape to cancel out the time-dependent Stark shift. StarkV has the duration of the risefall.

Reply:
StarkV has the duration of `round_pulse(2 * sigma * risefall)`. We can think of `2 * sigma * risefall` as the ideal width of StarkV. When it gets rounded, we can either stretch sigma by the same factor to keep `risefall` constant or we can keep `sigma` fix and let the stretched width redefine `risefall`. So either we redefine `sigma` to `sigma_dt` like you do or we redefine `risefall`. To do that, we would keep `sigma` fixed for both pulses but pass `ramps_dt / 2 / sigma` as `risefall_sigma_ratio` to StarkU. Keeping sigma fixed seems slightly better to me, but it probably is such a small difference that either is fine.

Reply:
Here is the code I was using to compare the cases.

```python
from matplotlib import pyplot as plt
from qiskit import pulse


amp = 1
sigma = 13
width = 30
risefall = 1.25

g_duration = 10 * round(risefall * sigma * 2 / 10)

risefall = g_duration / 2 / sigma
#sigma = g_duration / risefall / 2
print(risefall)
print(sigma)

g = pulse.Gaussian(g_duration, amp, sigma)
gs = pulse.GaussianSquare(g_duration + width, amp, sigma, risefall_sigma_ratio=risefall)

fig = plt.figure()
ax = fig.add_subplot()
ax.plot(g.get_waveform().samples, "o-")
ax.plot(gs.get_waveform().samples, "o-")
ax.set_xlim(0, g_duration / 2)
```

Reply:
Thanks. I decided to update `BackendTiming` which looks simpler. I was more concerning the pulse renormalization (lifted Gaussian) at small sigma since it modifies spectrum of the pulse, but these are usually tiny difference as you mention :)